### ISSUE TITLE
support framework attributes

### ISSUE THREAD

**Issue by PatrickJS** [validator/validator#1722]
Qwik uses `q:` for attributes
```html
<!doctype html>
<html
  q:container="paused"
  q:version="1.5.2-dev20240418142224"
  q:render="ssr-dev"
  q:base="/repl/1lsst7gz46v/build/"
  q:manifest-hash="vhqpfk"
>
  <!--qv q:key=0t_1--><head q:head>
    <title q:head>Hello Qwik</title>
  </head>
  <body>
    <!--qv q:id=0 q:key=Ncbm:0t_0-->
    <main q:key="4e_0">
      <h1>Qwik</h1>
      <section>
        <h2>Core features</h2>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
          eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
          minim veniam, quis nostrud exercitation ullamco laboris nisi ut
          aliquip ex ea commodo consequat.
        </p>
      </section>
    </main>
    <!--/qv-->
  </body>
  <!--/qv-->
  <script type="qwik/json">
    {"refs":{},"ctx":{},"objs":[],"subs":[]}
  </script>
</html>
```

**Comment by sideshowbarker** [comment]
Thanks for taking time to report this, but the tool in this repo is an HTML checker, for HTML documents that use standard HTML attributes and elements that exist in the HTML specification https://html.spec.whatwg.org and that are documented in MDN and other standard sources for documentation of the HTML language. It’s not a tool for checking documents that use non-standard attribute names.

**Comment by PatrickJS** [comment]
sadly there is a team that is being prevented from using modern frameworks since they require this checker to pass

**Comment by sideshowbarker** [comment]
I recognize there’s a legitimate problem here in that there’s a team somewhere which apparently believes that when they take some document that uses non-standard HTML attributes, and run it through a standards-based tool for checking HTML documents, the tool should somehow recognize their non-standard attributes and not report anything about them.

But it’s unclear why there’s an expectation that the issue tracker in this repo is the place to find a solution for that problem.

The entire purpose of the tool covered by this issue tracker is to check documents for conformance with the standard requirements in the HTML specification. And nowhere in the HTML specification are any attributes named `q:container` etc defined. So the tool covered by this issue tracker necessarily emits a message essentially saying, _“I found an attribute named `q:container` in this document, but there’s no attribute by the name defined in the HTML specification”_.

Anyone using a `q:container` attribute intentionally can safely choose to completely ignore any messages this tool reports about it — and further, they can use one of the many filtering options documented at https://github.com/validator/validator/wiki/Message-filtering to _persistently_ filter/drop/ignore/suppress the messages, so that they will never see them again.

Beyond that, it’s completely out of scope for this issue tracker to try to solve the apparent problem.

**Comment by PatrickJS** [comment]
ok I link the team with your comment. thanks

**Comment by speedyhoon** [comment]
> sadly there is a team that is being prevented from using modern frameworks since they require this checker to pass

In my opinion, Qwik should have been designed properly with [`data-*`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/data-*) instead of `q:*`.

**Comment by PatrickJS** [comment]
I agree
